### PR TITLE
Re-order

### PR DESCRIPTION
**Pull request by yasiloghmani**
Re-Order 

Description:
The PR adds Re-order component, Upgrade css  style and component structure. Add "Pagination" , "Sort By Date" and "Filter By Date" to re-order. Create "Orders details page" 

Linting:

- [X] No linting errors

Tests:

- [X] Manual tests

Documentation:

- [X] Requires documentation updates

### CODE REVIEW

File: src/OrdersHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  useEffect(() => {
+    filterByDate();
+  }, [filterByDate])
+
+  return (
+    <div className="ordershistory">
+      <h1 className="ordershistory__title">{t('orders')}</h1>
+      <div className="ordershistory__searchfilterbar">
+        <button className="ordershistory__searchfilterbar --mbldate" onClick={sortByDate}>
+                  <span>{t('date')}</span>
```
Comment (shaunmaharaj) [inline src/OrdersHistory.tsx:61]: formatting structure

**Comment by shaunmaharaj** [comment]
@yasiloghmani please update the PR title with more information, the description to clearly state what's implemented, and check necessary requirements in the PR template prior to us merging.

File: src/app-state.ts [hunk 2]
```diff
@@ -218,22 +218,28 @@ function usePurchaseHistoryState() {
...
+ 
   useEffect(() => {
-    if (token) {
-      getAllOrders(token).then((res: any) => {
+    if (token) {  
+      getAllOrders(token, pageNum).then((res: any) => {
         setData(res.data);
+        setTotal(res.meta.page.total);
+        setCurrentPage(res.meta.page.current)
+        console.log(res.meta)
```
Comment (shaunmaharaj) [inline src/app-state.ts:231]: leftover console.log

File: src/SideMenu.tsx [hunk 3]
```diff
@@ -40,14 +45,19 @@ export const SideMenu: React.FC = (props) => {
...
     <div className="sidemenu" ref={ref}>
       <button className="sidemenu__btn" onClick={handleSelectorClicked}>
-        {currentSideMenuItems.length > 0 && t(currentSideMenuItems[0].children)}
+        {currentSideMenuItems.length > 0 ? t(currentSideMenuItems[0].children) : t("orders")}
       </button>
       <div className={`sidemenu__dropdown ${!isOpen ? 'sidemenu__hidden' : ''}`}>
         {sideMenuItems.map(elem => (
           <div className='sidemenu__item' key={elem.to}>
-            <Link to={elem.to} className={`sidemenu__link ${location.pathname === elem.to ? '--selected' : ''}`} onClick={handleHideMenu}>{t(elem.children)}</Link>
+            {location.pathname === `/orderdetails/${orderId}`
```
Comment (shaunmaharaj) [inline src/SideMenu.tsx:53]: What was the reason for this change? It's the same link and action but the only difference is the classNames used. Readability?

File: src/SideMenu.tsx [hunk 4]
```diff
@@ -40,14 +45,19 @@ export const SideMenu: React.FC = (props) => {
   return (
     <div className="sidemenu" ref={ref}>
       <button className="sidemenu__btn" onClick={handleSelectorClicked}>
-        {currentSideMenuItems.length > 0 && t(currentSideMenuItems[0].children)}
+        {currentSideMenuItems.length > 0 ? t(currentSideMenuItems[0].children) : t("orders")}
```
Comment (shaunmaharaj) [inline src/SideMenu.tsx:48]: will there always be a currentSideMenuItems[0] element, or could this break in your new assumption?

Comment (yasiloghmani) [reply src/SideMenu.tsx:48]: I was trying to check 2 other pathnames, I think it is  a better idea to call the URL same as what we did for "addressUrl" and add it to the "sideMenuItems" arrays. What do you think?
Add : `const orderDetailsUrl = createOrderDetailsUrl()`
Update : `const sideMenuItems = [
    { to: accountUrl, children: 'my-account' },
    { to: addressUrl, children: 'addresses' },
    { to: ordersHistoryUrl, children: 'orders' }
{ to: orderDetailsUrl, children: 'orders' }
  ];`

and change he condition to : `<Link to={elem.to} className={`sidemenu__link ${location.pathname.includes(elem.to) ? '--selected' : ''}`} onClick={handleHideMenu}>{t(elem.children)}</Link>`

**Review by shaunmaharaj — APPROVED** [review]

**Pull request merged**